x/stake/client/rest/query.go

```diff
@@ -51,52 +118,147 @@ func delegationHandlerFn(cliCtx context.CLIContext, cdc *wire.Codec) http.Handle
+			// if we also would put in redelegations from, we would have every redelegation double
+			redelegations, statusCode, errMsg, err := getDelegatorRedelegations(cliCtx, cdc, delegatorAddr, validatorAddr)
+			if err != nil {
+				w.WriteHeader(statusCode)
+				w.Write([]byte(fmt.Sprintf("%s%s", errMsg, err.Error())))
+				return
+			}
+			if statusCode != http.StatusNoContent {
+				delegationSummary.Redelegations = append(delegationSummary.Redelegations, redelegations)
+			}
+
+			output, err := cdc.MarshalJSON(delegationSummary)
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				w.Write([]byte(err.Error()))
+				return
+			}
+
+			// success
+			w.Write(output) // write
+		}
+	}
+}
+
+// HTTP request handler to query all staking txs (msgs) from a delegator
+func delegatorTxsHandlerFn(cliCtx context.CLIContext, cdc *wire.Codec) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var output []byte
+		var typesQuerySlice []string
+		vars := mux.Vars(r)
+		delegatorAddr := vars["delegatorAddr"]
+
+		_, err := sdk.AccAddressFromBech32(delegatorAddr)
 		if err != nil {
 			w.WriteHeader(http.StatusBadRequest)
 			w.Write([]byte(err.Error()))
 			return
 		}
 
-		key := stake.GetDelegationKey(delegatorAddr, validatorAddr)
-
-		res, err := cliCtx.QueryStore(key, storeName)
+		node, err := cliCtx.GetNode()
 		if err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(fmt.Sprintf("couldn't query delegation. Error: %s", err.Error())))
+			w.Write([]byte(fmt.Sprintf("Couldn't get current Node information. Error: %s", err.Error())))
 			return
 		}
 
-		// the query will return empty if there is no data for this record
-		if len(res) == 0 {
+		// Get values from query
+
+		typesQuery := r.URL.Query().Get("type")
+		trimmedQuery := strings.TrimSpace(typesQuery)
+		if len(trimmedQuery) != 0 {
+			typesQuerySlice = strings.Split(trimmedQuery, " ")
+		}
+
+		noQuery := len(typesQuerySlice) == 0
+		isBondTx := contains(typesQuerySlice, "bond")
+		isUnbondTx := contains(typesQuerySlice, "unbond")
+		isRedTx := contains(typesQuerySlice, "redelegate")
+		var txs = []tx.Info{}
+		var actions []string
+
+		if isBondTx {
+			actions = append(actions, string(tags.ActionDelegate))
+		} else if isUnbondTx {
+			actions = append(actions, string(tags.ActionBeginUnbonding))
+			actions = append(actions, string(tags.ActionCompleteUnbonding))
+		} else if isRedTx {
+			actions = append(actions, string(tags.ActionBeginRedelegation))
+			actions = append(actions, string(tags.ActionCompleteRedelegation))
+		} else if noQuery {
+			actions = append(actions, string(tags.ActionDelegate))
+			actions = append(actions, string(tags.ActionBeginUnbonding))
+			actions = append(actions, string(tags.ActionCompleteUnbonding))
+			actions = append(actions, string(tags.ActionBeginRedelegation))
+			actions = append(actions, string(tags.ActionCompleteRedelegation))
+		} else {
```

Comment:
Use a switch statement here?

Reply:
Since it doesn't has an expression I thought it was better with `if`. But I can change it for better legibility